### PR TITLE
Add a streaming_op() to the pr_curves plugin.

### PR DESCRIPTION
**Pull request by agrubb**
This op produces the same result (a Tensor Summary containing the PR
curve data) as the existing op() function, but it does so by
aggregating the result over multiple batches of data.  The function
also returns an update_op that should be used to update the summary
for each batch of data.

With these changes, the function provides a tf.metrics-like function
for generating PR curve data.

### CODE REVIEW

File: tensorboard/plugins/pr_curve/summary.py [hunk 1]
```diff
@@ -164,6 +164,98 @@ def op(
...
+
+  Returns:
+    pr_curve: A string `Tensor` containing a single value: the
+      serialized PR curve Tensor summary. The summary contains a
+      float32 `Tensor` of dimension (6, num_thresholds). The first
+      dimension (of length 6) is of the order: true positives, false
+      positives, true negatives, false negatives, precision, recall.
+    update_op: An operation that updates the summary with the latest data.
+  """
+  thresholds = [i * 1.0 / float(num_thresholds - 1)
```
Comment (chihuahua) [inline tensorboard/plugins/pr_curve/summary.py]: Does `i / float(` ... suffice? Without the `* 1.0`?

File: tensorboard/plugins/pr_curve/summary_test.py [hunk 2]
```diff
@@ -320,5 +320,81 @@ def testRawDataOp(self):
...
+
+      proto = self.pb_via_op(pr_curve)
+      expected_proto = self.pb_via_op(expected_pr_curve)
+
+      # Need to detect and fix the automatic _1 appended to second namespace.
+      self.assertEqual(proto.value[0].tag, 'pr_curve/pr_curves')
+      self.assertEqual(expected_proto.value[0].tag, 'pr_curve_1/pr_curves')
+      expected_proto.value[0].tag = 'pr_curve/pr_curves'
+
+      self.assertProtoEquals(expected_proto, proto)
```
Comment (chihuahua) [inline tensorboard/plugins/pr_curve/summary_test.py]: Oh wow, I should use `assertProtoEquals` more often.

File: tensorboard/plugins/pr_curve/summary.py (hunk 1 above)
Comment (agrubb) [reply tensorboard/plugins/pr_curve/summary.py]: Yep. Done.

**Comment by agrubb** [comment]
Updated tensorboard/summary.py.  Also verified all tests pass (locally) with latest tf-nightly.

File: tensorboard/plugins/pr_curve/summary.py [hunk 3]
```diff
@@ -164,6 +164,98 @@ def op(
         description,
         collections)
 
+
+def streaming_op(tag, labels, predictions, num_thresholds=200, weights=None,
```
Comment (chihuahua) [inline tensorboard/plugins/pr_curve/summary.py]: nit: Could we reorganize the arguments to be one per row?

Comment (agrubb) [reply tensorboard/plugins/pr_curve/summary.py]: Done.

**Pull request merged**